src/main/java/io/jenkins/plugins/checks/github/GitHubSCMSourceStatusChecksProperties.java

```diff
@@ -0,0 +1,69 @@
+package io.jenkins.plugins.checks.github;
+
+import edu.hm.hafner.util.VisibleForTesting;
+import hudson.Extension;
+import hudson.model.Job;
+import io.jenkins.plugins.checks.status.StatusChecksProperties;
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMSource;
+
+import java.util.Optional;
+
+/**
+ * Implementing {@link StatusChecksProperties} to retrieve properties from jobs with
+ * {@link GitHubSCMSourceStatusChecksTrait}.
+ */
+@Extension
+public class GitHubSCMSourceStatusChecksProperties implements StatusChecksProperties {
+    private final SCMFacade scmFacade;
+
+    /**
+     * Default constructor.
+     */
+    public GitHubSCMSourceStatusChecksProperties() {
+        this(new SCMFacade());
+    }
+
+    @VisibleForTesting
+    GitHubSCMSourceStatusChecksProperties(final SCMFacade scmFacade) {
+        this.scmFacade = scmFacade;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
```

Comment:
Such comments can be removed since the IDE is smart enough

Reply:
I added more meaningful docs.